enterprise/views.py

```diff
@@ -160,6 +165,34 @@ def render_page_with_error_code_message(request, context_data, error_code, log_m
     )
 
 
+def get_create_ent_enrollment(
+        course_id,
+        enterprise_customer_user,
+):
+    """
+    Get or Create the Enterprise Course Enrollment.
+
+    This method will set the Source if the Enterprise Course Enrollment (ECE)
+    was already created. It is possible the ECE will have been created by the
+    background task at the time of enrollment so we need to update the source
+    in that case.
+    """
+    source = EnterpriseEnrollmentSource.get_source(EnterpriseEnrollmentSource.ENROLLMENT_URL)
+    # Create the Enterprise backend database records for this course
+    # enrollment
+    enterprise_course_enrollment, created = EnterpriseCourseEnrollment.objects.get_or_create(
+        enterprise_customer_user=enterprise_customer_user,
+        course_id=course_id,
+        defaults={
+            'source': source
+        }
+    )
+    if not created and enterprise_course_enrollment.source is None:
+        enterprise_course_enrollment.source = source
```

Comment:
i'm not sure if this should be set if it's empty; technically we don't know where it came from and i'm not sure if we should assume it is from an enrollment url.